pkg/preemptiontoleration/preemption_toleration_policy.go

```diff
@@ -0,0 +1,72 @@
+package preemptiontoleration
+
+import (
+	"strconv"
+
+	schedulingv1 "k8s.io/api/scheduling/v1"
+	"k8s.io/utils/pointer"
+)
+
+const (
+	AnnotationKeyPrefix                     = "preemption-toleration.scheduling.sigs.k8s.io/"
+	AnnotationKeyEnabled                    = AnnotationKeyPrefix + "enabled"
+	AnnotationKeyMinimumPreemptablePriority = AnnotationKeyPrefix + "minimum-preemptable-priority"
+	AnnotationKeyTolerationSeconds          = AnnotationKeyPrefix + "toleration-seconds"
+)
+
+// PreemptionTolerationPolicy holds preemption toleration policy configuration.  Each property values are annotated in the target PriorityClass resource.
+// Example:
+//   kind: PriorityClass
+//     metadata:
+//     name: toleration-policy-sample
+//     annotation:
+//       preemption-toleration.scheduling.sigs.k8s.io/enabled: ""
+//       preemption-toleration.scheduling.sigs.k8s.io/minimum-preemptable-priority: "10000"
+//       preemption-toleration.scheduling.sigs.k8s.io/toleration-seconds: "3600"
+type PreemptionTolerationPolicy struct {
+	// MinimumPreemptablePriority specifies the minimum priority value that can preempt this priority class.
+	// It defaults to the PriorityClass's priority value + 1 if not set, which means pods that have a higher priority value can preempt it.
+	MinimumPreemptablePriority *int32 `json:"minimumPreemptablePriority,omitempty"`
```

Comment:
> I feel like this should serve as a concrete value instead of pointer

it makes sense.  fixed in a537368d0dea5e862e4fb523d1dbe8f4f14dbc9f

> BTW: is the json tag mandatory?

No, actually.  fixed in e2c7bdfaf9d7fa76bf785fa53ca67fb4cde0955b